### DISCUSSION TITLE
How to use launch.json

### DISCUSSION THREAD

**Discussion by 3ynm** [emacs-lsp/lsp-mode#3475] [Q&A]
I've put this `launch.json` file on my project root:

```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "PHP Debug",
            "type": "php",
            "request": "launch",
            "program": "${file}",
            "args": ["--server=4711"],
            "cwd": "${workspaceFolder}",
            "pathMappings": {
                "/opt/hawese/core": "${workspaceFolder}"
            }
        }
    ]
}
```

But its config is not listed when I do `dap-debug` on my PHP project. I'm following this guide https://emacs-lsp.github.io/lsp-mode/tutorials/php-guide/#creating-debug-configuration

Am I doing something wrong?

**Comment by yyoncho** [comment]
Is this happening when using the config from the tutorial? (cc @elken )

**Reply by elken** [reply]
Ah, I made a mistake. It should appear under `.vscode`, I made an assumption. Will amend.

**Reply by yyoncho** [reply]
hm, the CPP tutorial is like that too. I think that dap-mode is supposed to pick launch.json even if it is in the project root.

**Reply by elken** [reply]
Hm, it seems like it does indeed check both `ROOT` and `ROOT/.vscode` (`dap-variables-find-vscode-config`)

**Comment by elken** [comment]
Hi @hacktivista, thanks for opening a discussion!

~~This file should be under `.vscode` in the project root, that one's on me so thanks for spotting it!~~

~~Is the [PR](https://github.com/emacs-lsp/lsp-mode/pull/3477) I created clear enough?~~

I tried to replicate your setup, and it worked fine for me. Can you try and produce a minimal working example? The below is what I setup
![image](https://user-images.githubusercontent.com/2872862/164170640-54ff9f2e-0318-4840-b0d0-e034f2de7ae4.png)
to give
![image](https://user-images.githubusercontent.com/2872862/164170711-f25030bd-9314-4cd7-b870-f38ee977e3e9.png)

**Reply by 3ynm** [reply]
For some reason I don't get that completions

![image](https://user-images.githubusercontent.com/5282749/164273447-f87c436a-ac1e-498d-8306-b6c6cf11db29.png)

I have `dap-mode` version `20220416.1252` installed and emacs `28.1`.

Maybe I could debug what's happening if you give me something to look at.

PD: Php Remote is an elisp config I add on my .emacs

**Reply by elken** [reply]
![image](https://user-images.githubusercontent.com/2872862/164388689-2faf9cf1-0d0a-411e-8031-f918a91b8446.png)

Using `emacs -Q -l emacs-php.el file.php` where `emacs-php.el` is the config from the tutorial, it still works. It looks like you're running your own config in the above screenshot (I can see evil in the modeline) so first thing would be try and replicate it without your config.

**Comment by elken** [comment]
@hacktivista any updates to this?
